extensions/amp-skimlinks/0.1/test/helpers.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpSkimlinks} from '../amp-skimlinks';
+import {CustomEventReporterBuilder} from '../../../../src/extension-analytics';
+import {Services} from '../../../../src/services';
+import {Tracking} from '../tracking';
+import {pubcode} from './constants';
+
+const helpersFactory = env => {
+  const {win} = env;
+
+  return {
+    createAnchor(href) {
+      const anchor = win.document.createElement('a');
+      anchor.href = href;
+
+      return anchor;
+    },
+
+    createStubXhr(data) {
+      const response = {
+        json: () => {
+          return Promise.resolve(data);
+        },
+      };
+
+      return {
+        fetchJson: env.sandbox.stub().returns(Promise.resolve(response)),
+      };
+    },
+
+    mockServiceGetter(getterName, returnValue) {
+      env.sandbox.stub(Services, getterName);
+      Services[getterName].returns(returnValue);
+    },
+
+    stubCustomEventReporterBuilder() {
```

Comment:
This is a completely rewrite of the `customEventReporterBuilder`, can we simply create a real builder instead of trying to stub every function that's involved? Same applies to other stub functions in testing.

Reply:
`CustomEventReporterBuilder` requires an `AmpElement` to initialize which is something I don't have in my tests. The stubbing of the whole `customEventReporterBuilder` class allows me to abstract the requirements and side effects of `customEventReporterBuilder` and focus only on what I need to test.